### PR TITLE
Add autoscaling metrics collection support

### LINKED ISSUE #25168: ec2_asg - hability to enable monitoring in autoscaling groups
##### ISSUE TYPE

 - Feature Idea

##### COMPONENT NAME

ec2_asg module
##### ANSIBLE VERSION

```
ansible 2.3.0.0
  config file = 
  configured module search path = Default w/o overrides
  python version = 2.7.9 (default, Jun 29 2016, 13:08:31) [GCC 4.9.2]
```

##### OS / ENVIRONMENT

Debian Jessie x86_84 on AWS instance
##### SUMMARY

Would be great if the module could enable monitoring for Autoscaling Groups. By now, each time a ASG is created, monitoring needs to be manually enabled.
##### STEPS TO REPRODUCE

Create an ASG with module ec2_asg

### PR DESCRIPTION
**Pull request by tomislacker**
##### SUMMARY
This addition adds support to [enable/disable ASG metrics collection](http://docs.aws.amazon.com/autoscaling/latest/userguide/as-instance-monitoring.html#as-enable-group-metrics), as well as picking which metrics should be collected.

Fixes #25168

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
`ec2_asg`

##### ANSIBLE VERSION

```
ansible 2.3.0.0 (detached HEAD d56ba0945d) last updated 2017/04/27 12:13:35 (GMT -500)
  config file = 
  configured module search path = Default w/o overrides
  python version = 2.7.10 (default, Nov 28 2015, 19:16:30) [GCC 4.8.5]
```

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/ec2_asg.py
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/ec2_asg.py]: Please remove `The only valid value is 1Minute.`

That is covered by the `choices:` line

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by tomislacker** [comment]
ready_for_review

**Comment by tomislacker** [comment]
ready_for_review
@gundalow I've made the requested change.

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
@tomislacker Thanks, looks good

Needs a review from people that know AWS, I believe @willthames is the maintainer for this module.

File: lib/ansible/modules/cloud/amazon/ec2_asg.py
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ec2_asg.py:828]: is the `any` required here - `if metrics_to_disable:` should suffice, I'd think

File: lib/ansible/modules/cloud/amazon/ec2_asg.py
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ec2_asg.py:806]: Please use [standard exception handling guidelines](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-2)

File: lib/ansible/modules/cloud/amazon/ec2_asg.py
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ec2_asg.py:184]: `required: False` is the default and so not required here

**Review by willthames — CHANGES_REQUESTED** [review]
This is all mostly good, just want to get the exception handling more standardised.

**Comment by jippi** [comment]
Any ETA for this PR ? :)

**Comment by levisbakalinsky** [comment]
+1

**Comment by ryansb** [comment]
Waiting on edits from @tomislacker

**Comment by SurenNihalani** [comment]
I think this is already merged in master in 404f9260d9e

**Comment by s-hertel** [comment]
Thanks @SurenNihalani 
Closing since this was added in 404f926

**Pull request closed without merging**
